interface/patient_file/encounter/forms.php

```diff
@@ -632,10 +638,11 @@ function divtoggle(spanid, divid) {
         echo "<div class='form_header_controls'>";
 
         // If the form is locked, it is no longer editable
-        if ( $esign->isLocked() ) {
-            echo "<a href=# class='css_button_small form-edit-button-locked' id='form-edit-button-".attr($formdir)."-".attr($iter['id'])."'><span>".xlt('Locked')."</span></a>";
-        } else {
-          if (!$aco_spec || acl_check($aco_spec[0], $aco_spec[1], '' , 'write')) {
+        if ($esign->isLocked()) {
+                 echo "<a href=# class='css_button_small form-edit-button-locked' id='form-edit-button-" . attr($formdir) . "-" . attr($iter['id']) . "'><span>" . xlt('Locked') . "</span></a>";
+          } else {
+          if ((!$aco_spec || acl_check($aco_spec[0], $aco_spec[1], '' , 'write') AND $GLOBALS['enable_group_therapy'] == 0)
+              OR (((!$aco_spec || acl_check($aco_spec[0], $aco_spec[1], '' , 'write')) AND $GLOBALS['enable_group_therapy'] AND acl_check("groups","glog",false, 'write')))) {
```

Comment:
Here it appears that if group therapy is enabled then all forms will require glog permission. What about forms that are not related to group therapy? And also, since we now have form-specific permissions why do you need any changes here at all?